apps/prairielearn/src/components/InstructorInfoPanel.html.ts

```diff
@@ -0,0 +1,372 @@
+          ${question.qid}
+        </a>
+      </div>
+    </div>
+
+    ${question_is_shared && course.sharing_name
+      ? html`
+          <div class="d-flex flex-wrap">
+            <div class="pr-1">Shared As:</div>
+            ${question.shared_publicly
+              ? html`
+                  <div>
+                    <a
+                      href="${config.urlPrefix}/public/course/${course.id}/question/${question.id}/preview"
+                    >
+                      @${course.sharing_name}/${question.qid}
+                    </a>
+                  </div>
+                `
+              : html`<div>@${course.sharing_name}/${question.qid}</div>`}
+          </div>
+        `
+      : ''}
+    <div class="d-flex flex-wrap">
+      <div class="pr-1">Title:</div>
+      <div>${question.title}</div>
+    </div>
+  `;
+}
+
+function VariantInfo({ variant }: { variant?: Variant & { formatted_date: string } }) {
+  if (variant == null) return '';
+
+  const duration =
+    // Some legacy queries still return the duration as a string, so parse it before formatting
+    typeof variant.duration === 'string'
+      ? IntervalSchema.parse(variant.duration)
+      : variant.duration ?? 0;
+  return html`
+    <div class="d-flex flex-wrap">
+      <div class="pr-1">Started at:</div>
+      <div>${variant.formatted_date}</div>
+    </div>
+    <div class="d-flex flex-wrap">
+      <div class="pr-1">Duration:</div>
+      <div>${formatInterval(duration)}</div>
+    </div>
+    <div class="d-flex flex-wrap pb-2">
+      <div class="pr-1">
+        <button class="btn btn-link" data-toggle="collapse" data-target="#instructorTrue_answer">
+          Show/Hide answer
+        </button>
+      </div>
+      <div class="collapse" id="instructorTrue_answer">
+        <code>${JSON.stringify(variant.true_answer)}</code>
+      </div>
+    </div>
+  `;
+}
+
+function AssessmentInstanceInfo({
+  assessment,
+  assessment_instance,
+}: {
+  assessment?: Assessment;
+  assessment_instance?: AssessmentInstance & { formatted_date: string };
+}) {
+  if (assessment == null || assessment_instance == null) return '';
+
+  const instructorUrlPrefix = `${config.urlPrefix}/course_instance/${assessment.course_instance_id}/instructor`;
+  const duration =
+    // Some legacy queries still return the duration as a string, so parse it before formatting
+    typeof assessment_instance.duration === 'string'
+      ? IntervalSchema.parse(assessment_instance.duration)
+      : assessment_instance.duration ?? 0;
+
+  return html`
+    <hr />
+    <h5 class="card-title">Assessment Instance:</h5>
+    <div class="d-flex flex-wrap">
+      <div class="pr-1">AID:</div>
+      <div>
+        <a href="${instructorUrlPrefix}/assessment/${assessment.id}">${assessment.tid}</a>
+      </div>
+    </div>
+
+    <div class="d-flex flex-wrap">
+      <div class="pr-1">Started at:</div>
+      <div>${assessment_instance.formatted_date}</div>
```

Comment:
Done in 91ba840ea. I have not changed whatever query is used to retrieve these values, as it would be tricky to make sure that these values are not used elsewhere, I'll leave that for when we have stronger typing in res.locals. Same for variant date.